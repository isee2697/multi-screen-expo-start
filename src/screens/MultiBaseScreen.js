import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { colors, utilStyles } from '../api/constants';

// grab component
import Touch from '../components/Touch';

const MultiBaseScreen = props => {
  const { navigation } = props;

  return (
    <View style={styles.container}>
      <ScrollView
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
      >
        <Text style={styles.textStyle}>Multi Screens Content Area</Text>

        <View style={utilStyles.spacer80} />

        <Touch
          onPress={() => navigation.navigate('MultiLevel2')}
          style={utilStyles.btn}
          text="go to level 2"
        />
      </ScrollView>
    </View>
  );
};

MultiBaseScreen.navigationOptions = {
  headerTitleStyle: {
    flex: 1,
    textAlign: 'center'
  },
  title: 'Multi Screens'
};

MultiBaseScreen.propTypes = {
  // required
  navigation: PropTypes.object.isRequired
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.brandPrimary,
    flex: 1
  },
  contentContainer: {
    alignItems: 'center',
    paddingTop: 30
  },
  textStyle: {
    color: colors.white
  }
});

export default MultiBaseScreen;
