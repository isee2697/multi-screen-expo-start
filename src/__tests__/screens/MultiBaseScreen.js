import React from 'react';
import { shallow } from 'enzyme';

import MultiBaseScreen from '../../screens/MultiBaseScreen';

const navigation = { navigate: jest.fn() };
const screen = shallow(<MultiBaseScreen navigation={navigation} />);

describe('<MultiBaseScreen />', () => {
  it('renders correctly', () => {
    expect(screen).toMatchSnapshot();
  });
});
