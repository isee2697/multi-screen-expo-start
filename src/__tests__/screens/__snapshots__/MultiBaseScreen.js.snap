// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiBaseScreen /> renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1d2d3a",
        }
      }
    >
      Multi Screens Content Area
    </Text>
    <Component
      style={
        Object {
          "height": 80,
          "width": "100%",
        }
      }
    />
    <Touch
      accesCType="button"
      accesLabel={null}
      accesTrait="button"
      accessible={true}
      icon={null}
      onPress={[Function]}
      opacityActive={0.8}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#1d2d3a",
          "borderRadius": 80,
          "borderWidth": 0.5,
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        }
      }
      text="go to level 2"
      textStyle={
        Object {
          "color": "#1d2d3a",
          "textAlign": "center",
        }
      }
    />
  </ScrollViewMock>
</Component>
`;
